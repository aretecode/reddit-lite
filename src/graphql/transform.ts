import { isObjPure, isArray } from '../utils/is'
import { EMPTY_OBJ, EMPTY_ARRAY } from '../utils/EMPTY'
import {
  UnsafeFetchMoreResponse,
  RedditLitePostsGraphQLResponse,
  RedditLitePostsResponse,
} from '../typings'

/**
 * @todo should not have to be done on client side
 */
export function fromFetchMoreToSafeUpdate(args: UnsafeFetchMoreResponse) {
  const { fetchMoreResult } = args

  const newPosts =
    isObjPure(fetchMoreResult) && isObjPure(fetchMoreResult.posts)
      ? fetchMoreResult.posts
      : EMPTY_OBJ

  const newList = isArray(newPosts.list) ? newPosts.list : EMPTY_ARRAY

  const newBefore = newPosts.before
  const newAfter = newPosts.after

  return {
    newPosts,
    newList,
    newBefore,
    newAfter,
  }
}

export function fromNewDataToApollo(newData: RedditLitePostsResponse) {
  return {
    posts: {
      ...newData,
      __typename: 'RedditLiteResponse',
    },
    __typename: '',
  } as RedditLitePostsGraphQLResponse
}
