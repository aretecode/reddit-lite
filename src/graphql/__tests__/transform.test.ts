import { fromFetchMoreToSafeUpdate, fromNewDataToApollo } from '../transform'

describe('graphql', () => {
  describe('transform', () => {
    it('should safely transform our response', () => {
      const transformed = fromFetchMoreToSafeUpdate({})
      const apollo = fromNewDataToApollo({})
      expect(transformed).toMatchSnapshot()
      expect(apollo).toMatchSnapshot()
    })
  })
})
