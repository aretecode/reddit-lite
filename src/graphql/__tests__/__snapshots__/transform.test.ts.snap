// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql transform should safely transform our response 1`] = `
Array [
  Object {
    "authorFlairText": "google searches for you",
    "authorFullName": "t2_3rnmf",
    "body": "Are you looking for something to do? Would you like to advertise your event? Want to suggest something to everyone? Whether it's a show, a play, a conference, a restaurant opening, or whatever, this is where you can stick all of it. This is a monthly post that is useful for Vancouverites, useful for tourists, and useful for the main Vancouver sub.

**New**

Please post any opportunities to volunteer that you want the subreddit to know about. This thread will be curated periodically to keep the events of the week fresh and visible.

**Who?**

Events don't need to be yours! You're welcome to post any interesting/fun/unknown/well known/moderately-enjoyable events. Will remove posts completely off topic (like a link to one's unrelated blog or comments about how there's nothing to do in this city)

**Tip**

Sort the comments below BY NEW to see the latest events


[April Events Thread](https://www.reddit.com/r/vancouver/comments/b5dlhh/april_events_and_volunteering_thread_please_post/)",
    "commentCount": 50,
    "createdAtIso": "2019-04-30T20:55:32-0700",
    "createdAtPretty": "14 days",
    "createdAtUtc": 1556682932,
    "id": "bjcurx",
    "imageUrl": "",
    "isSticky": true,
    "postKind": "text",
    "score": 23,
    "title": "May Events and Volunteering Thread. Please post your upcoming events and volunteering opportunities here.",
    "url": "https://reddit.com/r/vancouver/comments/bjcurx/may_events_and_volunteering_thread_please_post/",
  },
  Object {
    "authorFlairText": "",
    "authorFullName": "t2_fhc4s",
    "body": "",
    "commentCount": 97,
    "createdAtIso": "2019-05-14T07:52:29-0700",
    "createdAtPretty": "about 12 hours",
    "createdAtUtc": 1557845549,
    "id": "bojx2w",
    "imageUrl": "https://b.thumbs.redditmedia.com/2fNRoJSzaYCxg_wBJJYcmAdqD5sZnWk0sRUN1YVM-RM.jpg",
    "isSticky": false,
    "postKind": "image",
    "score": 301,
    "title": "Spotted in Coquitlam this morning . Be careful out there",
    "url": "https://reddit.com/r/vancouver/comments/bojx2w/spotted_in_coquitlam_this_morning_be_careful_out/",
  },
  Object {
    "authorFlairText": "",
    "authorFullName": "t2_s9mls",
    "body": "",
    "commentCount": 60,
    "createdAtIso": "2019-05-14T05:53:24-0700",
    "createdAtPretty": "about 14 hours",
    "createdAtUtc": 1557838404,
    "id": "boijsv",
    "imageUrl": "https://b.thumbs.redditmedia.com/dhYQ706l8zOp01lkS4MntVLQaV8TrgQf31UTI_LwDjU.jpg",
    "isSticky": false,
    "postKind": "link",
    "score": 189,
    "title": "Extensive bus-only lanes coming to 41st Avenue for new B-Line rapid bus",
    "url": "https://reddit.com/r/vancouver/comments/boijsv/extensive_busonly_lanes_coming_to_41st_avenue_for/",
  },
]
`;
